src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1947 @@
+        if (not df.at[person_id, 'is_alive']) or (df.at[person_id, 'age_exact_years'] >= 5):
+            # or
+            # df.at[person_id, 'un_am_treatment_type'].isin(['standard_RUTF', 'soy_RUSF', 'CSB++', 'inpatient_care']):
+            return
+
+        def schedule_next_monitoring():
+            def get_monitoring_frequency_days(age):
+                # TODO: in future maybe 0-1 to be dealt with within epi module
+                if age < 1:
+                    return p['growth_monitoring_frequency_days_agecat'][0]
+                elif age <= 2:
+                    return p['growth_monitoring_frequency_days_agecat'][1]
+                else:
+                    return p['growth_monitoring_frequency_days_agecat'][2]
+
+            person_monitoring_frequency = get_monitoring_frequency_days(df.at[person_id, 'age_exact_years'])
+            if (df.at[person_id, 'age_exact_years'] + (person_monitoring_frequency / 365.25)) < 5:
+                # schedule next growth monitoring
+                self.sim.modules['HealthSystem'].schedule_hsi_event(
+                    hsi_event=HSI_Wasting_GrowthMonitoring(module=self.module, person_id=person_id),
+                    topen=self.sim.date + pd.DateOffset(days=person_monitoring_frequency),
+                    tclose=None,
+                    priority=2
+                )
+            # else: no more growth monitoring scheduled as the age will be above 5 at the time
+
+        # TODO: as stated above, for now we schedule next monitoring for all children, even those sent for treatment
+        schedule_next_monitoring()
+
+        # but if they are currently treated, the growth monitoring will not go through
+        # TODO: later could be scheduled for monitoring within the tx to use the resources
+        if (df.at[person_id, 'un_last_wasting_date_of_onset'] < df.at[person_id, 'un_am_tx_start_date'] <
+                self.sim.date):
+            return
+        # or if HSI was already scheduled due to care-seeking, no need to attend the growth monitoring
+        hsi_event_scheduled = [
+            ev
+            for ev in self.sim.modules["HealthSystem"].find_events_for_person(person_id)
+            if isinstance(ev[1], (HSI_Wasting_SupplementaryFeedingProgramme_MAM,
+                                  HSI_Wasting_OutpatientTherapeuticProgramme_SAM,
+                                  HSI_Wasting_InpatientTherapeuticCare_ComplicatedSAM))
+        ]
+        if hsi_event_scheduled:
+            return
+
+        # the person may not attend the appt
+        if not self.attendance:
+            return
+
+        available_equipment = []
+        for equip in ['Height Pole (Stadiometer)', 'Weighing scale', 'MUAC tape']:
+            available = rng.random_sample() < HSI_Event.probability_all_equipment_available(self, equip)
+            if available:
+                available_equipment.append(equip)
+        self.add_equipment(set(available_equipment))
```

Comment:
please declare all equipment -- not conditional on whether it is available or not.

if anything is so essential that the HSI could not proceed without it, declare it in the init.